dask/array/optimization.py

```diff
@@ -41,13 +41,17 @@ def optimize(
     if isinstance(dsk, HighLevelGraph):
         dsk = optimize_blockwise(dsk, keys=keys)
         dsk = fuse_roots(dsk, keys=keys)
+        dsk = cull_highlevelgraph(dsk, keys)
+        dsk2 = ensure_dict(dsk)
+        dependencies = {k: get_dependencies(dsk2, k, as_list=True) for k in dsk2}
+    else:
+        dsk = ensure_dict(dsk)
+        dsk2, dependencies = cull(dsk, keys)
```

Comment:
Does this case ever occur?  Do any arrays not use HLGs?

Reply:
Nope :)